internal/controller/services/monitoring/resources/prometheus-route.tmpl.yaml

```diff
@@ -3,12 +3,23 @@ apiVersion: route.openshift.io/v1
 metadata:
   name: data-science-prometheus-route
   namespace: {{.Namespace}}
+  labels:
+    app: namespace-restricted-metrics
+    platform.opendatahub.io/part-of: monitoring
+  ownerReferences:
+    - apiVersion: {{.Component.APIVersion}}
+      kind: {{.Component.Kind}}
+      name: {{.Component.Name}}
+      uid: {{.Component.UID}}
 spec:
   path: /
   to:
     kind: Service
-    name: prometheus-operated
+    name: namespace-restricted-metrics
```

Comment:
I think this should still be prometheus so it's clear what it routes to. Even `prometheus-restricted` if we want to include that it is restricted in some way.